jbpm-human-task/jbpm-human-task-workitems/src/test/java/org/jbpm/services/task/wih/HTWorkItemHandlerBaseTest.java

```diff
@@ -608,8 +610,142 @@ public void testTaskWithoutCreatedBy() throws Exception {
+        ksession.setEnvironment(env);
+        ksession.setWorkItemManager(manager);
+        WorkItemImpl workItem = new WorkItemImpl();
+        workItem.setName("Human Task");
+        workItem.setParameter("NodeName", "TaskName");
+        workItem.setParameter("Comment", "Comment");
+        workItem.setParameter("Priority", "10");
+        workItem.setParameter("GroupId", "Crusaders");
+        workItem.setParameter("ActorId", "Darth Vader");
+        workItem.setParameter("SwimlaneActorId", "Darth Vader");
+        workItem.setProcessInstanceId(10);
+        handler.executeWorkItem(workItem, manager);
+
+        
+        List<TaskSummary> tasks = taskService.getTasksAssignedAsPotentialOwner("Darth Vader", "en-UK");
+        assertEquals(1, tasks.size());
+        TaskSummary task = tasks.get(0);
+        assertEquals("TaskName", task.getName());
+        assertEquals(10, task.getPriority().intValue());
+        assertEquals("Comment", task.getDescription());
+        assertEquals(Status.Ready, task.getStatus());
+        assertEquals(10, task.getProcessInstanceId().intValue());
+
+        taskService.start(task.getId(), "Darth Vader");
+        taskService.complete(task.getId(), "Darth Vader", null);
+
+        assertTrue(manager.waitTillCompleted(MANAGER_COMPLETION_WAIT_TIME));
+    }
+    
+    @Test
+    public void testTaskWithDisableAutoClaimTaskWithGroupOnly() throws Exception {
+        TestWorkItemManager manager = new TestWorkItemManager();
+        env.set("Autoclaim", "false");
+        ksession.setEnvironment(env);
+        ksession.setWorkItemManager(manager);
+        WorkItemImpl workItem = new WorkItemImpl();
+        workItem.setName("Human Task");
+        workItem.setParameter("NodeName", "TaskName");
+        workItem.setParameter("Comment", "Comment");
+        workItem.setParameter("Priority", "10");
+        workItem.setParameter("GroupId", "Crusaders");
+        workItem.setParameter("SwimlaneActorId", "Darth Vader");
+        workItem.setProcessInstanceId(10);
+        handler.executeWorkItem(workItem, manager);
+
+        
+        List<TaskSummary> tasks = taskService.getTasksAssignedAsPotentialOwner("Darth Vader", "en-UK");
+        assertEquals(1, tasks.size());
+        TaskSummary task = tasks.get(0);
+        assertEquals("TaskName", task.getName());
+        assertEquals(10, task.getPriority().intValue());
+        assertEquals("Comment", task.getDescription());
+        assertEquals(Status.Ready, task.getStatus());
+        assertEquals(10, task.getProcessInstanceId().intValue());
+
+        taskService.start(task.getId(), "Darth Vader");
+        taskService.complete(task.getId(), "Darth Vader", null);
+
+        assertTrue(manager.waitTillCompleted(MANAGER_COMPLETION_WAIT_TIME));
+    }
+    
+    @Test
+    public void testTaskWithAutoClaimTaskWithActorAndGroup() throws Exception {
         TestWorkItemManager manager = new TestWorkItemManager();
+        env.set("Autoclaim", null);
```

Comment:
Would you mind removing this manual setting to null? Environment will produce null if nothing with that key is set anyway. Thanks!

Reply:
Done

Reply:
Ah, sorry, I meant it together with setEnvironment call :) I will express myself more clearly next time.

Reply:
Don't worry :) I should have deleted the setEnviroment too. Done!